src/OnchainKitProvider.tsx

```diff
@@ -48,6 +56,32 @@ export function OnchainKitProvider({
     return onchainKitConfig;
   }, [address, apiKey, chain, config, projectId, rpcUrl, schemaId]);
 
+  // Check the context for WagmiProvider
+  // Wagmi configuration defaults to the provided config if it exists
+  // Otherwise, use the OnchainKit-provided Wagmi configuration
+  let providedConfig = null;
+  try {
+    providedConfig = useConfig();
+  } catch (error) {
+    if (!(error instanceof WagmiProviderNotFoundError)) {
+      console.error('Error fetching config, using OnchainKit defaults:', error);
+      throw error;
+    }
+  }
+
+  // If WagmiProvider is not found, return the context with defaulted parent providers
+  if (!providedConfig) {
+    return (
+      <WagmiProvider config={getDefaultConfig({ apiKey, config })}>
```

Comment:
could you pass onchainkticonfig.config here or move the defaults to a consts file so they aren't duplicated?